src/sentry/api/serializers/rest_framework/dashboard.py

```diff
@@ -305,6 +318,40 @@ def validate(self, data):
                     has_query_error = True
                 else:
                     query_errors.append({})
+
+                if not query.get("on_demand_extraction_disabled", False):
+                    if query.get("columns"):
+                        all_columns = all_columns.union(query.get("columns"))
+                    # If this query wants ondemand check if we'll go over spec
+                    organization = self.context["organization"]
+                    widget_query = DashboardWidgetQuery(
+                        fields=query["fields"],
+                        aggregates=query.get("aggregates"),
+                        columns=query.get("columns"),
+                        field_aliases=query.get("field_aliases"),
+                        conditions=query["conditions"],
+                        name=query.get("name", ""),
+                        orderby=query.get("orderby", ""),
+                    )
+                    # Get widget specs if we haven't yet
+                    if current_widget_specs is None:
+                        current_widget_specs = get_current_widget_specs(organization)
+                    widget_specs = _get_widget_on_demand_specs(widget_query, organization)
+                    if len(widget_specs) == 0:
+                        # In the case the query errors, we should still disbale
+                        self.query_warnings["queries"].append(
+                            OnDemandExtractionState.DISABLED_QUERY_ERROR
+                        )
```

Comment:
OnDemandExtractionState.DISABLED_QUERY_ERROR -> `DISABLED_NOT_APPLICABLE` since it's an expected case that we have **0** specs; MEP data covers some queries already in which case a spec isn't created.

It's possible we don't need this either, since we should be checking again during the save whether it has specs, and if it doesn't, saving as `NOT_APPLICABLE` then too.